ql/src/java/org/apache/hadoop/hive/ql/exec/CommonMergeJoinOperator.java

```diff
@@ -92,6 +94,7 @@
 
   // A field because we cannot multi-inherit.
   transient InterruptibleProcessing interruptChecker;
+  transient boolean shortcutExtensionRows;
```

Comment:
Minor: should we use `unmatched` everywhere we use `extension`?

"Unmatched rows of an outer join" sounds more natural than "Extension rows of an outer join".